### PR TITLE
PS-376 Implement WatchForChanges method

### PR DESCRIPTION
**Pull request by lindseysimple**
Implement the WatchForChanges method.

Signed-off-by: Lindsey Cheng <beckysocute@gmail.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-configuration/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: internal/pkg/keeper/client.go [hunk 1]
```diff
@@ -140,9 +143,80 @@ func (client *keeperClient) GetConfiguration(configStruct interface{}) (interfac
...
+
+	messages := make(chan msgTypes.MessageEnvelope)
+	topic := path.Join(keeperTopicPrefix, client.configBasePath, waitKey, "#")
+	topics := []msgTypes.TopicChannel{
+		{
+			Topic:    topic,
+			Messages: messages,
+		},
+	}
+	msgBusConfig := config.(*models.ConfigurationStruct).MessageQueue
```
Comment (cloudxxx8) [inline internal/pkg/keeper/client.go]: always check error for type conversion

File: internal/pkg/keeper/client.go [hunk 2]
```diff
@@ -140,9 +143,80 @@ func (client *keeperClient) GetConfiguration(configStruct interface{}) (interfac
...
+	messageBus, err := messaging.NewMessageClient(msgTypes.MessageBusConfig{
+		SubscribeHost: msgTypes.HostInfo{
+			Host:     msgBusConfig.Host,
+			Port:     msgBusConfig.Port,
+			Protocol: msgBusConfig.Protocol,
+		},
+		Type:     msgBusConfig.Type,
+		Optional: msgBusConfig.Optional,
```
Comment (cloudxxx8) [inline internal/pkg/keeper/client.go:193]: add a deepCopy func to produce a copy of messageBusConfig, and add a random number suffix in the MqttClientId option.
reference:
https://github.com/IOTechSystems/edgex-go-private/pull/314/files
https://github.com/IOTechSystems/edgex-go-private/blob/b95546cea1628225578aae0059b0751de3f401c8/internal/support/sparkplug/mqtt/client.go#L249-L252

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: internal/pkg/keeper/client.go (hunk 1 above)
Comment (lindseysimple) [reply internal/pkg/keeper/client.go]: Done

File: internal/pkg/keeper/client.go (hunk 2 above)
Comment (lindseysimple) [reply internal/pkg/keeper/client.go:193]: Random number suffix in the MqttClientId added, thanks.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**